scipy/stats/stats.py

```diff
@@ -5819,7 +5689,185 @@ def chisquare(f_obs, f_exp=None, ddof=0, axis=0):
                             lambda_="pearson")
 
 
-Ks_2sampResult = namedtuple('Ks_2sampResult', ('statistic', 'pvalue'))
+KstestResult = namedtuple('KstestResult', ('statistic', 'pvalue'))
+
+
+def _compute_dplus(cdfvals):
+    """Computes D+ as used in the Kolmogorov-Smirnov test.
+
+    Parameters
+    ----------
+    cdfvals: array_like
+      Sorted array of CDF values between 0 and 1
+
+    Returns
+    -------
+      Maximum distance of the CDF values below Uniform(0, 1)
+"""
+    n = len(cdfvals)
+    return (np.arange(1.0, n + 1) / n - cdfvals).max()
+
+
+def _compute_dminus(cdfvals):
+    """Computes D- as used in the Kolmogorov-Smirnov test.
+
+    Parameters
+    ----------
+    cdfvals: array_like
+      Sorted array of CDF values between 0 and 1
+
+    Returns
+    -------
+      Maximum distance of the CDF values above Uniform(0, 1)
+    """
+    n = len(cdfvals)
+    return (cdfvals - np.arange(0.0, n)/n).max()
+
+
+def ks_1samp(x, cdf, args=(), alternative='two-sided', mode='auto'):
+    """
+    Performs the Kolmogorov-Smirnov test for goodness of fit.
+
+    This performs a test of the distribution F(x) of an observed
+    random variable against a given distribution G(x). Under the null
+    hypothesis, the two distributions are identical, F(x)=G(x). The
+    alternative hypothesis can be either 'two-sided' (default), 'less'
+    or 'greater'. The KS test is only valid for continuous distributions.
+
+    Parameters
+    ----------
+    x : array_like
+        a 1-D array of observations of random variables.
+    cdf : callable
```

Comment:
Right now `ks_1samp` doesn't accept a string.  `kstest(x, ...)` does, and it converts the string to a cdf function before passing down to `ks_1samp`.  Since `ks_1samp` is not in the API currently, it doesn't need to support all the myriad combinations of `kstest`.
If one wants the convenience of strings, use the wrapper `kstest` which figures out everything.